src/main/java/com/blackducksoftware/integration/hub/alert/digest/DigestNotificationProcessor.java

```diff
@@ -51,74 +58,115 @@
-                events.add(new EmailEvent(projectData));
-                events.add(new HipChatEvent(projectData));
-            });
-            return events;
+            final Collection<UserNotificationWrapper> userData = createCateoryDataMap(digestType, processedNotificationList);
+            return eventManager.createChannelEvents(userData);
         }
     }
 
-    private Collection<ProjectData> createCateoryDataMap(final DigestTypeEnum digestType, final Collection<NotificationEntity> eventMap) {
-        final Map<String, ProjectDataBuilder> projectDataMap = new LinkedHashMap<>();
-        for (final NotificationEntity entry : eventMap) {
-            final String projectKey = entry.getEventKey();
-            // get category map from the project or create the project data if
-            // it doesn't exist
-            Map<NotificationCategoryEnum, CategoryDataBuilder> categoryBuilderMap;
-            if (!projectDataMap.containsKey(projectKey)) {
-                final ProjectDataBuilder projectBuilder = new ProjectDataBuilder();
-                projectBuilder.setDigestType(digestType);
-                projectBuilder.setProjectName(entry.getProjectName());
-                projectBuilder.setProjectVersion(entry.getProjectVersion());
-                projectDataMap.put(projectKey, projectBuilder);
-                categoryBuilderMap = projectBuilder.getCategoryBuilderMap();
-            } else {
-                final ProjectDataBuilder projectBuilder = projectDataMap.get(projectKey);
-                categoryBuilderMap = projectBuilder.getCategoryBuilderMap();
-            }
-            // get the category data object to be able to add items.
-            CategoryDataBuilder categoryData;
-            final NotificationCategoryEnum categoryKey = NotificationCategoryEnum.valueOf(entry.getNotificationType());
-            if (!categoryBuilderMap.containsKey(categoryKey)) {
-                categoryData = new CategoryDataBuilder();
-                categoryData.setCategoryKey(categoryKey.name());
-                categoryBuilderMap.put(categoryKey, categoryData);
-            } else {
-                categoryData = categoryBuilderMap.get(categoryKey);
-            }
-            int count = 1;
-            final Map<String, Object> dataSet = new HashMap<>();
-            if (categoryKey == NotificationCategoryEnum.HIGH_VULNERABILITY || categoryKey == NotificationCategoryEnum.MEDIUM_VULNERABILITY || categoryKey == NotificationCategoryEnum.LOW_VULNERABILITY) {
-                count = entry.getVulnerabilityList().size();
-                dataSet.put(VulnerabilityCache.VULNERABILITY_ID_SET, getVulnerabilityIdSet(entry));
-            }
+    private Collection<UserNotificationWrapper> createCateoryDataMap(final DigestTypeEnum digestType, final Collection<NotificationEntity> notificationList) {
+        final Map<String, Map<String, ProjectDataBuilder>> userProjectMap = createUserProjectMap(digestType, notificationList);
```

Comment:
Instead of having maps in maps would it be better to create a new class and add functionality there instead?